src/app/frontend/deploy/validprotocol_directive.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** The name of this directive. */
+export const validProtocolValidationKey = 'validProtocol';
+
+/**
+ * Validates that chosen protocol is valid for chosen service type.
+ *
+ * @param {!angular.$resource} $resource
+ * @param {!angular.$q} $q
+ * @return {!angular.Directive}
+ * @ngInject
+ */
+export default function validProtocolDirective($resource, $q) {
+  const isExternalParam = 'isExternal';
+
+  return {
+    restrict: 'A',
+    require: 'ngModel',
+    scope: {
+      [isExternalParam]: '=',
+    },
+    link: function(scope, element, attrs, ctrl) {
+      /** @type {!angular.NgModelController} */
+      let ngModelController = ctrl;
+      let externalChanged = false;
+
+      scope.$watch(isExternalParam, (newVal, oldVal) => {
+        externalChanged = newVal !== oldVal;
```

Comment:
It will be triggered on isExternal but there will be no call to the backend for validation. I will give an example:

Initially we have `TCP` and `false` set. That is ok we do not to validate that as we assume that initial config is correct. User then clicks on external service switch so we have `TCP` and `true`. If for some reason backend logic will change and `TCP` and `true` will no longer be a valid combination then validation will not be fired until user touches protocol combobox.